source/extensions/transport_sockets/tls/ssl_socket.h

```diff
@@ -101,6 +109,7 @@ class SslSocket : public Network::TransportSocket,
   mutable std::string cached_sha_256_peer_certificate_digest_;
   mutable std::string cached_url_encoded_pem_encoded_peer_certificate_;
   mutable std::string cached_url_encoded_pem_encoded_peer_cert_chain_;
+  bool async_handshake_in_progress_{};
```

Comment:
Now `SslSocket` has 3 boolean flags `async_handshake_in_progress_`, `handshake_complete_`, `shutdown_sent_`.

@PiotrSikora @lizan would it make sense to have `SslSocketState` enum instead?

Reply:
Yeah that seems cleaner if works.